### PR TITLE
Added straightening rule

### PR DESCRIPTION
**Pull request by alex-e-leon**
@wooorm This PR adds a new rule that I've called straightening - which allows you to format tables without any kind of alignment (see the tests for examples).

This style is useful for preventing large diffs during single line changes.

Feel free to make any comments on the code or totally refactor it - I wrote it not knowing your preferences in regards to code-style, performance, etc. but would be keen to get this feature added.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -169,7 +169,7 @@ function markdownTable(table, options) {
 
     index = -1;
 
-    while (++index < cellCount) {
+    while (settings.straighten !== false && ++index < cellCount) {
```
Comment (wooorm) [inline index.js]: Could you wrap the whole `while` in an `if (settings.straighten !== false)`? No need to compute this on every iteration.

File: index.js [hunk 2]
```diff
@@ -204,11 +204,24 @@ function markdownTable(table, options) {
     rule = [];
 
     while (++index < cellCount) {
+      /* When `straighten` is false, make the rule the same size as the first row. */
+      if (settings.straighten === false) {
+        value = table[0][index];
+        if (value === null || value === undefined) {
```
Comment (wooorm) [inline index.js]: Could this `if/else` be replaced with a ternary operator? Something like: `value = value ? String(value) : ''`, or even `value = String(value || '')`

File: index.js [hunk 3]
```diff
@@ -204,11 +204,24 @@ function markdownTable(table, options) {
...
+          value = '';
+        } else {
+          value = String(value);
+        }
+        spacing = value.length > 3 ? value.length : 3;
```
Comment (wooorm) [inline index.js]: Could you add the `3` as a constant in the root scope? Something like `var MAX_FOO = 3`, together with `spacing = value.length > MAX_FOO ? value.length : MAX_FOO;`

File: readme.md [hunk 4]
```diff
@@ -116,6 +116,9 @@ The following options are available:
 *   `options.stringLength` ([`Function`][length], default:
     `s => s.length`)
     — Method to detect the length of a cell (see below).
+*   `options.straighten` (`boolean`, default: `true`)
+    — Whether to make all the table cells the same width. Setting this
```
Comment (wooorm) [inline readme.md]: Could you add a space after the sentence?

```diff
- width. Setting
+ width.  Setting
```

File: test.js [hunk 5]
```diff
@@ -182,6 +182,36 @@ test('table()', function (t) {
...
+      ['staging', 'fedcba9876543210']
+    ], {straighten: false}),
+    [
+      '| A |  |',
+      '| --- | --- |',
+      '| master | 0123456789abcdef |',
+      '| staging | fedcba9876543210 |'
+    ].join('\n'),
+    'handles short rules and missing elements for tables without straightening'
+  );
```
Comment (wooorm) [inline test.js]: Could you add a test for empty cells in the table body?

File: index.js (hunk 1 above)
Comment (alex-e-leon) [inline index.js]: That was my original approach but the only problem when doing this is that I received a warning from the linter about too many nested blocks.

I thought about re-factoring the code quite heavily to move some of the logic out into self contained functions, but didn't want to re-factor it too much without knowing what you thought.

File: index.js (hunk 2 above)
Comment (alex-e-leon) [inline index.js]: Sure, I actually copied this from a section further up - I should probably factor this out into a re-usable function TBH

File: index.js (hunk 3 above)
Comment (alex-e-leon) [inline index.js]: :+1:

File: readme.md (hunk 4 above)
Comment (alex-e-leon) [inline readme.md]: :+1:

File: test.js (hunk 5 above)
Comment (alex-e-leon) [inline test.js]: :+1:

File: index.js (hunk 1 above)
Comment (wooorm) [inline index.js]: Ah, darn linter. Ignore it. You can turn the rule off [here](https://github.com/wooorm/markdown-table/blob/db2909455918aa3752110d97ac541c919dae6bf3/package.json#L57).

> I thought about re-factoring the code quite heavily to move some of the logic out into self contained functions, but didn't want to re-factor it too much without knowing what you thought.

I’m OK with that, but I’d rather have that in another PR!

File: index.js (hunk 2 above)
Comment (wooorm) [inline index.js]: :+1:

**Comment by alex-e-leon** [comment]
@woorm I've updated the PR with the changes requested, do you mind taking a look and letting me know if it looks ok?

**Comment by wooorm** [comment]
Looks good! I’m wondering about one thing though: the name. Why not `straight: false`, or `align: false`, or `pretty: false`?

**Comment by alex-e-leon** [comment]
@wooorm good question! I didn't like the name `straighten` either TBH, but I couldn't think of a better term. `align` was my first thought, but it's already taken for`text-align` maybe `align-cells` or `straight`? I don't think `pretty` is a good term though. What are your thoughts?

**Comment by wooorm** [comment]
Does it make sense to pass both `straighten: false` and `align: 'l'` (and the like)? If it doesn’t, I’m OK with `align: false` (if not, we need a test case for that too)

**Comment by alex-e-leon** [comment]
That's a good point - we _could_ still use `align`, but I'm not sure I like the idea of overloading the meaning. If align actually means text-align, what does it mean when it's false? and it's weird that it could be either a string or a boolean value. I think I still prefer either `straight` or `align-cells` to differentiate it.

**Comment by wooorm** [comment]
You’re right! I just though of a case where passing `align: 'c'` (and the like) can work together with this new option, as `align` is also for the rendered markdown (such as when on GitHub).

How about `pad: false`?

**Comment by alex-e-leon** [comment]
:+1: Thats the best one yet. Will change it to that

**Comment by alex-e-leon** [comment]
@wooorm Updated so it's now `pad`
When this gets merged, do you mind adding the option to remark-stringify as well (or if you prefer I can try dive into remark-stringify and open a PR)

**Comment by wooorm** [comment]
Released as 1.1.0. Thanks a lot @alex-e-leon, pleasure working with you!

Would you be interested in PRing an option into `remark-stringify`?
